pkg/apis/core/validation/validation.go

```diff
@@ -1770,20 +1770,30 @@ func ValidatePersistentVolume(pv *core.PersistentVolume) field.ErrorList {
 // ValidatePersistentVolumeUpdate tests to see if the update is legal for an end user to make.
```

Comment:
add name and namespace verification for the new secret field to validateCSIPersistentVolumeSource, just like the other secret fields (if present, name and namespace are required and must fit secret and namespace name validation)